### PR TITLE
Catch duplicate run name error and throw custom exception

### PR DESCRIPTION
**Pull request by schustmi**
## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to the documentation, I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Describe changes
- Catch duplicate run name error and throw custom exception

### CODE REVIEW

File: src/zenml/orchestrators/utils.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        return "%dd%dh%dm%ds" % (days, hours, minutes, int_seconds)
+    elif hours > 0:
+        return "%dh%dm%ds" % (hours, minutes, int_seconds)
+    elif minutes > 0:
+        return "%dm%ds" % (minutes, int_seconds)
+    else:
+        return f"{seconds:.3f}s"
+
+
+def execute_tfx_component(
```
Comment (htahir1) [inline src/zenml/orchestrators/utils.py:35]: Can we call this just `execute_component` or `execute_step` instead? ;-)

File: src/zenml/orchestrators/utils.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+from tfx.orchestration.portable import data_types, launcher
+
+from zenml.exceptions import DuplicateRunNameError
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+def _format_timedelta_pretty(seconds: float) -> str:
```
Comment (htahir1) [inline src/zenml/orchestrators/utils.py:12]: I think we can put this somewhere that is more accessible?

Comment (htahir1) [reply src/zenml/orchestrators/utils.py:12]: I think `cli/utis.py` has a similar method. maybe merge these and put in general utils ?

File: src/zenml/orchestrators/utils.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    Returns:
+        Optional execution info returned by the launcher.
+    """
+    step_name = tfx_launcher._pipeline_node.node_info.id  # type: ignore[attr-defined] # noqa
+    start_time = time.time()
+    logger.info(f"Step `{step_name}` has started.")
+    try:
+        execution_info = tfx_launcher.launch()
+    except RuntimeError as e:
+        if "execution has already succeeded" in str(e):
```
Comment (htahir1) [inline src/zenml/orchestrators/utils.py:52]: haha! no TODO?

File: src/zenml/exceptions.py [hunk 4]
```diff
@@ -156,3 +156,14 @@ def __init__(
 
 class IntegrationError(Exception):
     """Raises exceptions when a requested integration can not be activated."""
+
+
+class DuplicateRunNameError(RuntimeError):
+    """Raises exception when a run with the same name already exists."""
```
Comment (htahir1) [inline src/zenml/exceptions.py:162]: Maybe we can call this a `DuplicateError` instead to generalize to other Duplicate use-cases?

**Review by htahir1 — COMMENTED** [review]
Cheers!

File: src/zenml/orchestrators/utils.py (hunk 3 above)
Comment (schustmi) [reply src/zenml/orchestrators/utils.py:52]: Haha well if you got any suggestion on what should be done to improve it I'm happy to add a todo :P

File: src/zenml/exceptions.py (hunk 4 above)
Comment (schustmi) [reply src/zenml/exceptions.py:162]: Duplicate in general or some kind of Metadata store duplicate? `DuplicateError` sounds a little too generic to me, but I agree that like this it's maybe a little too specific.

File: src/zenml/orchestrators/utils.py (hunk 2 above)
Comment (schustmi) [reply src/zenml/orchestrators/utils.py:12]: zenml/utils/logging_utils? zenml/utils/formatting_utils? zenml/utils/string_utils? Any other suggestion?

File: src/zenml/orchestrators/utils.py (hunk 3 above)
Comment (htahir1) [reply src/zenml/orchestrators/utils.py:52]: No forget the todo and let it burn us in the future! :-D

File: src/zenml/exceptions.py (hunk 4 above)
Comment (htahir1) [reply src/zenml/exceptions.py:162]: I was thinking maybe there are other duplicates like `DuplicateStepName` or something. You can make scoped to the Metadata store if you'd like

File: src/zenml/orchestrators/utils.py (hunk 2 above)
Comment (htahir1) [reply src/zenml/orchestrators/utils.py:12]: ` zenml/utils/string_utils`  sounds good

File: src/zenml/exceptions.py (hunk 4 above)
Comment (schustmi) [reply src/zenml/exceptions.py:162]: What do you think of `DuplicateContextError`? Then it can sort of refer to anything in the metadata store?

Comment (htahir1) [reply src/zenml/exceptions.py:162]: Won't that conflict with the notion of a `StepContext`?

Comment (michael-zenml) [reply src/zenml/exceptions.py:162]: Mhm yeah, any other suggestions that aren't as broad as `DuplicateError`?

Comment (htahir1) [reply src/zenml/exceptions.py:162]: Lets leave it like this and merge

**Pull request merged**